packages/react/src/components/Button/Button.js

```diff
@@ -143,19 +143,33 @@ const Button = React.forwardRef(function Button(
   let otherProps = {
     disabled,
     type,
+    'aria-describedby': kind === 'danger' ? 'danger-description' : null,
     'aria-pressed': hasIconOnly && kind === 'ghost' ? isSelected : null,
   };
   const anchorProps = {
     href,
   };
-  const assistiveText = hasIconOnly ? (
-    <div
-      ref={tooltipRef}
-      onMouseEnter={handleMouseEnter}
-      className={`${prefix}--assistive-text`}>
-      {iconDescription}
-    </div>
-  ) : null;
+
+  let assistiveText;
+  if (hasIconOnly) {
+    assistiveText = (
+      <div
+        ref={tooltipRef}
+        onMouseEnter={handleMouseEnter}
+        className={`${prefix}--assistive-text`}>
+        {iconDescription}
+      </div>
+    );
+  } else if (kind === 'danger') {
+    assistiveText = (
+      <span id="danger-description" className={`${prefix}--visually-hidden`}>
+        danger
```

Comment:
Would this need to be configurable / translatable?

Reply:
allowing it to be configurable via props (and thus translatable) is smart!